src/dialog/Dialog.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  TextStyle,
+  StyleProp,
+  ViewStyle,
+  ActivityIndicatorProps,
+  ActivityIndicator,
+  Platform,
+} from 'react-native';
+import Overlay, { OverlayProps } from '../overlay/Overlay';
+import Button, { ButtonProps } from '../buttons/Button';
+import { Theme } from '../config/theme';
+import { withTheme } from '../config';
+import { TextProps } from '../text/Text';
+
+export type DialogProps = Omit<OverlayProps, 'fullScreen'> & {
+  loading?: boolean;
+  loadingStyle?: StyleProp<ViewStyle>;
+  loadingProps?: ActivityIndicatorProps;
+  title?: string;
+  titleStyle?: StyleProp<TextStyle>;
+  titleProps?: TextProps;
+  noButtons?: boolean;
+  primary?: string;
+  primaryOnPress?(): void;
+  primaryButtonProps?: ButtonProps;
+  secondary?: string;
+  secondaryOnPress?(): void;
+  secondaryButtonProps?: ButtonProps;
+  theme?: Theme;
+};
+
+const Dialog: React.FunctionComponent<DialogProps> = ({
+  children,
+  loading,
```

Comment:
I think I would prefer the other method such as in `listItem` where you create composable pieces rather than the giant monolith components. So it would be better to have a `Dialog.Loading` and `Dialog.Title`, so that we aren't overburdened with props, confusing logic in the component, and people asking for more and more injection spots. Perhaps rework the component with this strategy more in mind.